cmd/agent/app/reporter/grpc/builder_test.go

```diff
@@ -308,12 +315,10 @@ func TestProxyClientTLS(t *testing.T) {
 			defer cancel()
 			var opts []grpc.ServerOption
 			if test.serverTLS.Enabled {
-				tlsCfg, err := test.serverTLS.Config(zap.NewNop())
+				tlsCfg, err := test.serverTLS.ToOtelServerConfig().LoadTLSConfig(ctx)
```

Comment:
can we change the test case struct itself to hold `configtls.ServerConfig` so that we don't have to do this conversion here?

Reply:
we could, but it would be just unnecessary changes in the PR. We're still reliant on the tlscfg structs in the main code. Plus, this is legacy agent code we don't even use anymore.